### ISSUE TITLE
Wrong filter parentheses when combining two "or" queries

### ISSUE THREAD

**Issue by zivillian** [object/Simple.OData.Client#209]
When two 'or' ODataExpressions are combined using the && operator, the resulting query has only parentheses around the first 'or' group.

```
var leftor = new ODataExpression<int>(x=>false) || new ODataExpression<int>(x=>x > 10);
var rightor = new ODataExpression<long>(x => false) || new ODataExpression<long>(x => x < -10);
var result = leftor && rightor;
```

results in something like `(false or x gt 10) and false or x lt -10`

My current workaround is an additional constant expression:

`
var result = leftor && new ODataExpression(Expression.Constant(true)) && rightor;
`
which results in `(false or x gt 10) and true and (false or x lt -10)`

The bug lies in [ODataExpression.Format.cs](https://github.com/object/Simple.OData.Client/blob/master/Simple.OData.Client.Core/Expressions/ODataExpression.Format.cs#L66), there is no check, wether both sides needs grouping, so only the left side is grouped by parentheses.

**Comment by object** [comment]
Thanks for the bug report. I will work on the fix.

**Comment by object** [comment]
Fixed in 4.17.1.
